azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/AzureComputeServiceAdapter.java

```diff
@@ -0,0 +1,429 @@
+         map.put( image.offer + "/" + image.sku, image.location);
+      }
+
+      for (VMImage image : images) {
+         image.globallyAvailable = map.get(image.offer + "/" + image.sku).containsAll(locations);
+      }
+   }
+
+   @Override
+   public VMImage getImage(final String id) {
+      String[] fields = VMImageToImage.decodeFieldsFromUniqueId(id);
+
+      Iterable<VMImage> images = listImages();
+
+      for (VMImage image : images) {
+         String imageId = VMImageToImage.encodeFieldsToUniqueId(image);
+         if (id.equals(imageId)){
+            return image;
+         }
+      }
+      return null;
+   }
+
+   @Override
+   public Iterable<Location> listLocations() {
+      List<Location> locations = api.getLocationApi().list();
+
+      List<ResourceProviderMetaData> resources = api.getResourceProviderApi().get("Microsoft.Compute");
+
+      final List<String> vmLocations = new ArrayList<String>();
+
+      for (ResourceProviderMetaData m : resources){
+         if (m.resourceType().equals("virtualMachines")){
+            vmLocations.addAll(m.locations());
+            break;
+         }
+      }
+
+      Iterable<Location> result = Iterables.filter(locations, new Predicate<Location>() {
+         @Override
+         public boolean apply(Location input) {
+            return vmLocations.contains(input.displayName());
+         }
+      });
+
+      return  result;
+   }
+
+   private String getResourceGroupFromId(String id) {
+      String searchStr = "/resourceGroups/";
+      int indexStart = id.lastIndexOf(searchStr) + searchStr.length();
+      searchStr = "/providers/";
+      int indexEnd = id.lastIndexOf(searchStr);
+
+      String resourceGroup = id.substring(indexStart, indexEnd);
+      return resourceGroup;
+   }
+
+   @Override
+   public VMDeployment getNode(final String id) {
+      Deployment deployment = api.getDeploymentApi(azureGroup).get(id);
+      if (deployment == null)
+         return null;
+      String resourceGroup = getResourceGroupFromId(deployment.id());
+      VMDeployment vmDeployment = new VMDeployment();
+      vmDeployment.deployment = deployment;
+      List<PublicIPAddress> list = getIPAddresses(deployment);
+      vmDeployment.ipAddressList = list;
+      return vmDeployment;
+   }
+
+   @Override
+   public void destroyNode(final String id) {
+      logger.debug("Destroying %s ...", id);
+      String storageAccountName = id.replaceAll("[^A-Za-z0-9 ]", "") + "storage";
+      int index = id.lastIndexOf("-");
+      String group = id.substring(0, index);
+
+      // Delete VM
+      URI uri = api.getVirtualMachineApi(azureGroup).delete(id);
+      if (uri != null){
+         boolean jobDone = Predicates2.retry(new Predicate<URI>() {
+            @Override public boolean apply(URI uri) {
+               return ParseJobStatus.JobStatus.DONE == api.getJobApi().jobStatus(uri);
+            }
+         }, 60 * 10 * 1000 /* 5 minute timeout */).apply(uri);
```

Comment:
Refactor this into a reusable predicate (see https://github.com/jclouds/jclouds-labs/pull/280#discussion_r65962712). This will also allow users to configure these timeouts and polling.